oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/optimizer/OptimizerService.kt

```diff
@@ -0,0 +1,13 @@
+package com.empowerops.volition.ref_oasis.optimizer
+
+class OptimizerService(
+        private val startAction: IStartAction,
+        private val stopAction: IStopAction,
+        private val forceStopAction: IForceStopAction,
+        private val pauseAction: IPauseAction,
+        private val resumeAction: IResumeAction
+) : IStartAction by startAction,
+        IStopAction by stopAction,
+        IForceStopAction by forceStopAction,
+        IPauseAction by pauseAction,
+        IResumeAction by resumeAction
```

Comment:
This is a C# convention not typically used in java. 

In java we're supposed to use adjectives for interface names. Thus, many communities (including kotlin all the way up to version 0.9) called them _traits_, not `interface`s.

So `IStopAction` becomes _`Stoppable`_ and `IForceStopAction` becomes _`ForceStoppable`_ and `IResumeAction` becomes `Resumable` etc.

Reply:
I dont know that I really like the java convention since it puses your skills pretty hard and it is often difficult to come up with the right adjectives, but it is the convention.